enf-reducer/cut-unnecessary-and.metta

```diff
@@ -0,0 +1,28 @@
+;(: AndCut (-> Expression Expression))
+(= (AndCut $expr)
+    (let*
+        (
+            (($gSet $child) (getGsetAndChildren $expr))
+            ($tail (cdr-atom $child))
+        )
+        (if (and (== $tail ()) (== $gSet ()))
+            (let*
+                (
+                    (($Grandgset $GrandChildren) (getGsetAndChildren (car-atom $child)))
+                    ($allLiterals (collapse (getGuardSet (superpose $GrandChildren) ())))
+                    ($test (collapse (let $a (getGsetAndChildren (superpose $GrandChildren)) (car-atom $a))))
+                    ($common (findCommon $allLiterals))
```

Comment:
This is just a questions. Why are you searching for common literals here?

Reply:
intersection of the guard sets of the children of C must be `empty` inorder to apply the transformation